app/javascript/packs/base.jsx

```diff
@@ -4,6 +4,35 @@ import {
   getInstantClick,
   initializeTouchDevice,
 } from '../topNavigation/utilities';
+import { fetchSearch } from '@utilities/search';
+
+// Namespace for functions which need to be accessed in plain JS initializers
+window.Forem = {};
+
+window.Forem.initializeMentionAutocompleteTextArea = async (
+  originalTextArea,
+) => {
+  const parentContainer = originalTextArea.parentElement;
+
+  const alreadyInitialized = parentContainer.id === 'combobox-container';
+  if (alreadyInitialized) {
+    return;
+  }
+
+  const [{ MentionAutocompleteTextArea }, { render, h }] = await Promise.all([
+    import('@crayons/MentionAutocompleteTextArea'),
+    import('preact'),
```

Comment:
I don't know about so I ask: does this mean that we potentially load `preact` multiple times in the same page? See `window.showModal` that also loads `preact`

Reply:
my understanding (@nickytonline correct me if wrong!) is that once a module has been imported once, any subsequent call to import the same module is effectively skipped, and you get a reference to the originally imported module 🙂

Reply:
my understanding (@nickytonline correct me if wrong!) is that once a module has been imported once, any subsequent call to import the same module is effectively skipped, and you get a reference to the originally imported module 🙂

Reply:
It will be in the browser cache at this point so it will make a request, but it will be cached in the browser, so pretty much instantaneous.

If someone uses the mention autocomplete and then a modal, it will fetch twice (but cached). We could optimize if it becomes an issue so that we only ever import dynamically once though.